hil/cli/project.py

```diff
@@ -80,6 +82,42 @@ def project_connect_node(project, node):
 @project_node.command(name='remove')
 @click.argument('project')
 @click.argument('node')
-def project_detach_node(project, node):
+@click.option('--force', is_flag=True)
+def project_detach_node(project, node, force):
     """Remove <node> from <project>"""
+    if force:
+        client.node.disable_obm(node)
+        print("Disabled OBM")
+        node_info = client.node.show(node)
+        statuses = []
+        if 'nics' in node_info:
+            for n in node_info['nics']:
+                # if any nic is connected to a network, then it has a switch
+                # and port. So call revert port on those.
+                if 'networks' in n:
+                    try:
+                        response = client.port.port_revert(
+                                                    n['switch'], n['port'])
+                        statuses.append(response['status_id'])
+                    except KeyError:
+                        print("Only admins can force remove nodes.")
+                        sys.exit(1)
```

Comment:
I'm not a huge fan of this; at the very least we should have a comment here explaining why a `KeyError` suggests admin-only, but in a case like this where we're doing inference as to what actually went wrong, I'd rather have the error message reflect what actually happend, and maybe some guesses -- it could get easy for something else to unexpectedly trigger this and have a user be very confused about the admin bit, since it's not actually an auth check that's triggering it.

Reply:
okay, I have put in a comment and the exit message has more details.